src/lib/censorMarriage.js

```diff
@@ -0,0 +1,134 @@
+const censorMarriage = (marriage, roles) => {
+  const censorField = (redact, value) => {
+    return marriage.status.blocked !== false || (redact && !roles.includes('full-details')) ?
+      null : value
+  }
```

Comment:
It's not a big deal but this could be re-factored into a shared function. You would need a closure though, something like:
```js
  const censor = blocked => (redact, value) => {
    return blocked !== false || (redact && !roles.includes('full-details')) ?
      null : value
  }
```
Then in `censorMarriages` you could create your function with the following:
```js
const censorField = censor(marriage.status.blocked)
```

I also wonder whether these `censorBirth`, `censorDeath`, `censorMarriage` functions belong together in one file.

Reply:
Added/extracted reusable censor function